src/train/train.py

```diff
@@ -0,0 +1,158 @@
+        point_wise_loss_func = task.get_metric()
+        point_wise_loss = point_wise_loss_func(output, ys.cuda()).mean(dim=0)
+
+        baseline_loss = (
+            sum(
+                max(curriculum.n_dims_truncated - ii, 0)
+                for ii in range(curriculum.n_points)
+            )
+            / curriculum.n_points
+        )
+
+        if i % args.wandb.log_every_steps == 0 and not args.test_run:
+            wandb.log(
+                {
+                    "overall_loss": loss,
+                    "excess_loss": loss / baseline_loss,
+                    "pointwise/loss": dict(
+                        zip(point_wise_tags, point_wise_loss.cpu().numpy())
+                    ),
+                    "n_points": curriculum.n_points,
+                    "n_dims": curriculum.n_dims_truncated,
+                },
+                step=i,
+            )
+
+        curriculum.update()
+
+        pbar.set_description(f"loss {loss}")
+        if i % args.training.save_every_steps == 0 and not args.test_run:
+            training_state = {
+                "model_state_dict": model.state_dict(),
+                "optimizer_state_dict": optimizer.state_dict(),
+                "train_step": i,
+            }
+            torch.save(training_state, state_path)
+
+        if (
+            args.training.keep_every_steps > 0
+            and i % args.training.keep_every_steps == 0
+            and not args.test_run
+            and i > 0
+        ):
+            torch.save(model.state_dict(), os.path.join(args.out_dir, f"model_{i}.pt"))
+
+
+def main(args):
+    if args.test_run:
+        curriculum_args = args.training.curriculum
+        curriculum_args.points.start = curriculum_args.points.end
+        curriculum_args.dims.start = curriculum_args.dims.end
+        args.training.train_steps = 100
+    else:
+        wandb.init(
+            dir=args.out_dir,
+            project=args.wandb.project,
+            entity=args.wandb.entity,
+            config=args.__dict__,
+            notes=args.wandb.notes,
+            name=args.wandb.name,
+            resume=True,
+        )
+
+    model = build_model(args.model)
+    model.cuda()
+    model.train()
+
+    train(model, args)
+
+    if not args.test_run:
+        _ = get_run_metrics(args.out_dir)  # precompute metrics for eval
+
+
+if __name__ == "__main__":
+    parser = QuinineArgumentParser(schema=schema)
+    args = parser.parse_quinfig()
+    assert args.model.family in ["gpt2", "lstm"]
+    print(f"Running with: {args}")
+
+    if not args.test_run:
+        run_id = args.training.resume_id
+        if run_id is None:
+            run_id = str(uuid.uuid4())
+
+        out_dir = os.path.join(args.out_dir, run_id)
+        if not os.path.exists(out_dir):
+            os.makedirs(out_dir)
+        args.out_dir = out_dir
+
+        with open(os.path.join(out_dir, "config.yaml"), "w") as yaml_file:
+            yaml.dump(args.__dict__, yaml_file, default_flow_style=False)
```

Comment:
This is very appealing as an approach to overriding parameters from CLI, although we should decide if that's even a good idea

Reply:
We aren't going to pursue this, since it might harm reproducibility if we don't fully merge the YAML config with CLI params.